### PR TITLE
Add support for JWT authentication against SalesForce

### PR DESCRIPTION
**Pull request by ehtycuact**
Followed the provided authentication mechanism to now work with JWT and key. 
Added tests for the new auth method. 
Fixed existing failing tests.

### CODE REVIEW

**Comment by ehtycuact** [comment]
@g8rswimmer Hey, I cannot seem to add you as a reviewer to this so I figured I should just mention you here. Cheers!

File: credentials/jwt.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+type claims struct {
+	jwt.StandardClaims
+}
+
+func (provider *jwtProvider) Retrieve() (io.Reader, error) {
+	expirationTime := provider.GetAppropriateExpirationTime()
+	tokenString, err := provider.BuildClaimsToken(expirationTime, provider.creds.URL, provider.creds.ClientId, provider.creds.ClientUsername)
+	if err != nil {
+		return nil, err
```
Comment (g8rswimmer) [inline credentials/jwt.go:23]: `suggestion`

I have started using error wrapping to add more information and it has really helped in debugging.

File: credentials/jwt.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+	return strings.NewReader(form.Encode()), nil
+}
+
+func (provider *jwtProvider) URL() string {
+	return provider.creds.URL
+}
+
+func (provider *jwtProvider) GetAppropriateExpirationTime() int64 {
+	return time.Now().Add(3 * time.Minute).Unix()
```
Comment (g8rswimmer) [inline credentials/jwt.go:38]: is there any data behind the 3 minute time?  Might want to make the timeout a constant for naming.

**Review by g8rswimmer — CHANGES_REQUESTED** [review]
there are only some comments are readability of the code, other than that looks pretty good

File: credentials/jwt.go (hunk 2 above)
Comment (ehtycuact) [reply credentials/jwt.go:38]: @g8rswimmer Hey! The 3 minute was a suggestion from some developer from SalesForce in their forum. Though, I cannot find the thread right now. I'll move it to a constant either way.

File: credentials/credentials.go [hunk 3]
```diff
@@ -98,3 +117,12 @@ func validatePasswordCredentials(cred PasswordCredentials) error {
 	}
 	return nil
 }
+
+func validateJWTCredentials(cred JwtCredentials) error {
```
[the remark this answers is not in the document]
Comment (ehtycuact) [reply credentials/credentials.go:121]: @g8rswimmer Sure thing. I changed your password validation method to a `switch` statement too.

**Comment by ehtycuact** [comment]
@g8rswimmer I've addressed your comments. Feel free to, please, review :) Cheers!

**Review by g8rswimmer — APPROVED** [review]

**Comment by ethycuact** [comment]
@g8rswimmer Hey! Would you mind smashing a release for this please?

**Pull request merged**